imap_data_access/file_validation.py

```diff
@@ -444,11 +539,142 @@ def construct_path(self) -> Path:
+        value = components[part]
+        year_int = int(value)
+        components[part] = int(year_int)
+        try:
+            components["start_date"] = datetime(year_int, 1, 1)
+        except ValueError:
+            return "Invalid year. Please ensure the year is not 0."
+        del components[part]
+
+    @staticmethod
+    def _handle_version(part, components):
+        """Parse version into an integer."""
+        value = components[part]
+        components[part] = int(value)
+
+    @staticmethod
+    def _spice_parts_handler(components):
+        """Validate and transform SPICE file compents.
+
+        Parameters
+        ----------
+        components : dict
+            Dictionary containing components of the file.
+
+        Returns
+        -------
+        components : dict
+            Dictionary containing components, validated and transformed.
+        """
+        parts_list = list(components.keys())
+        transforms = {
+            "start_year_doy": SPICEFilePath._handle_doy,
+            "end_year_doy": SPICEFilePath._handle_doy,
+            "start_date": SPICEFilePath._handle_date,
+            "end_date": SPICEFilePath._handle_date,
+            "type": SPICEFilePath._handle_type,
+            "start_year": SPICEFilePath._handle_start_year,
+            "version": SPICEFilePath._handle_version,
+        }
+        errors = ""
+        for part in parts_list:
+            if part in transforms:
+                err = transforms[part](part, components)
+                if err:
+                    errors += err
+        if errors:
+            raise SPICEFilePath.InvalidSPICEFileError(errors)
+        return components
+
+    @staticmethod
+    def extract_filename_components(filename: Path | str) -> dict | None:
+        """Extract all components from filename.
+
+        Will return a dictionary in the form:
+            version - int
+            type - string
+            extension - string
+            start_date - datetime
+            end_date - datetime
+
+        If a match is not found, InvalidSPICEFileError will be raised.
+
+        Parameters
+        ----------
+        filename : Path | str
+            The filename to parse
+
+        Returns
+        -------
+        components : dict
+            Dictionary containing components.
+        """
+        filename = Path(filename)
+        spice_metadata = None
+        for regex in SPICEFilePath.valid_spice_regexes:
+            m = regex.match(filename.name.lower())
+            if m is not None:
+                spice_metadata = SPICEFilePath._spice_parts_handler(m.groupdict())
+                spice_metadata["extension"] = filename.suffix[1:]
+                if "start_date" not in spice_metadata:
+                    spice_metadata["start_date"] = SPICEFilePath.EARLIEST_VALID_TIME
+                if "end_date" not in spice_metadata:
+                    spice_metadata["end_date"] = SPICEFilePath.LATEST_VALID_TIME
+                break
```

Comment:
Outside this loop you can handle the bare raise_case without needing the if then.
```suggestion
                return spice_metadata
```